Tests/Common/Util/ExtensionsTests.cs

```diff
@@ -771,6 +772,47 @@ public void DecimalTruncateTo3DecimalPlaces()
             Assert.AreEqual(10.999m, value.TruncateTo3DecimalPlaces());
         }
 
+        [Test]
+        public void TimeoutAfterAndContinueTests()
+        {
+            Action action = () => Thread.Sleep(1000);
+            Action divideByZeroAction = () => { var x = 0; var y = 1 / x; };
+
+            action.TimeoutAfterAndContinue(TimeSpan.FromMilliseconds(2000))
+                .ContinueWith(x => Assert.IsTrue(x.IsCompleted))
+                .Wait();
+
+            // Timeout exception: wait 900 ms for a 2s work
+            var task1 = action.TimeoutAfterAndContinue(TimeSpan.FromMilliseconds(900));
+            task1.ContinueWith(x =>
+                {
+                    Assert.IsTrue(x.IsFaulted);
+                    Assert.AreEqual(x.Exception?.InnerException?.GetType(), typeof(TimeoutException));
+                }
+            );
+            Assert.Throws<AggregateException>(() => task1.Wait());
+
+            // Action throws exception
+            var task2 = divideByZeroAction.TimeoutAfterAndContinue(TimeSpan.FromMilliseconds(900));
+            task2.ContinueWith(x =>
+                {
+                    Assert.IsTrue(x.IsFaulted);
+                    Assert.AreEqual(x.Exception?.InnerException?.GetType(), typeof(DivideByZeroException));
+                }
+            );
+            Assert.Throws<AggregateException>(() => task2.Wait());
+
+            // Callback throws exception
+            var task3 = action.TimeoutAfterAndContinue(TimeSpan.FromMilliseconds(2000), divideByZeroAction);
+            task3.ContinueWith(x =>
+                {
+                    Assert.IsTrue(x.IsFaulted);
+                    Assert.AreEqual(x.Exception?.InnerException?.GetType(), typeof(DivideByZeroException));
+                }
+            );
+            Assert.Throws<AggregateException>(() => task3.Wait());
```

Comment:
I think a nice unit test policy is to divide the different assertions into specific unit tests